File: bot/commands/registerClanCommand.py
from discord.client import Client
from requests import status_codes
from apiClient import ApiClient
from typing import Match
from discord.message import Message
from command_spec import MessageCommand, PhraseRepository
import re
import requests


class RegisterClanCommand(MessageCommand):
    def __init__(self, phraseRepository: PhraseRepository, apiClient: ApiClient) -> None:
        super().__init__(phraseRepository)
        self.apiClient = apiClient

    def isMatchTo(self, message: str):
        return re.search(self.phraseRepository.get('register_clan'), message)

    async def execute(self, message: Message) -> None:
        print('start register clan command')
        messageText: str = re.sub(
            f"@{self.phraseRepository.get('bot_name')}", '', message.clean_content).strip(" ")
        matches: Match = re.search(self.phraseRepository.get(
            'register_clan'), messageText)
        clanName: str = matches.group('clanName')

        response = self.apiClient.register_clan(clanName)

        if(response.status_code == 200):
            await message.add_reaction(self.phraseRepository.get('succeed_reaction'))
        elif(response.status_code == 400):
            await message.channel.send(response.text)
        else:
            await message.add_reaction(self.phraseRepository.get('failed_reaction'))

File: bot-js/src/commands/RegisterClanCommand.ts
import { Client, Message } from 'discord.js';
import { MessageCommand } from './MessageCommand';
import { PhraseRepository } from '../support/PhraseRepository';
import { ApiClient } from '../backend/ApiClient';
import { mentionedToMe } from '../Sylph';
import { PhraseKey } from '../support/PhraseKey';

export class RegisterClanCommand implements MessageCommand {
    constructor(
        private phraseRepository: PhraseRepository,
        private discordClient: Client,
        private apiClient: ApiClient,
    ) {
        this.commandPattern = new RegExp(this.phraseRepository.get(PhraseKey.registerClan()));
    }

    private readonly commandPattern: RegExp;

    isMatchTo(message: Message): Promise<boolean> {
        return Promise.resolve(
            this.commandPattern.test(message.cleanContent)
            && mentionedToMe(message, this.discordClient)
        );
    }

    async execute(message: Message): Promise<void> {
        console.log("start register clan command");
        const matches = this.commandPattern.exec(message.cleanContent);
        if (matches && matches.groups) {
            const name = matches.groups["clanName"];

            await this.apiClient.registerClan(name);

            await message.react(this.phraseRepository.get(PhraseKey.succeedReaction()));
        }
    }
}